environment/deployments/roundtable/main.tf

```diff
@@ -267,6 +267,82 @@ resource "google_service_account" "reaper_sa" {
   project      = module.project_factory.project_id
 }
 
+# WorkloadIdentityPool for workloads to use Workload Identity Federation
+# instead of maintaining long-lived service account creds.
+resource "google_iam_workload_identity_pool" "pool" {
+  workload_identity_pool_id = "main"
+  display_name              = "Main"
+  description               = "Main identity pool for this project"
+  project                   = module.project_factory.project_id
+}
+
+# A Workload Identity Provider for workloads running from GitHub actions.
+#
+# https://cloud.google.com/blog/products/identity-security/enabling-keyless-authentication-from-github-actions
+#
+# Available GitHub OIDC claims to map:
+# https://docs.github.com/en/actions/security-for-github-actions/security-hardening-your-deployments/about-security-hardening-with-openid-connect#understanding-the-oidc-token
+resource "google_iam_workload_identity_pool_provider" "github" {
+  workload_identity_pool_id          = google_iam_workload_identity_pool.pool.workload_identity_pool_id
+  workload_identity_pool_provider_id = "github"
+  display_name                       = "GitHub"
+  description                        = "GitHub Actions identity pool provider."
+  attribute_mapping = {
+    "google.subject"       = "assertion.sub"
+    "attribute.actor"      = "assertion.actor"
+    "attribute.aud"        = "assertion.aud"
+    "attribute.repository" = "assertion.repository"
+  }
+  oidc {
+    issuer_uri = "https://token.actions.githubusercontent.com"
+  }
+}
+
+locals {
+  full_prodromos_repo = "lsst-sqre/prodromos"
+}
+
+# Service account for Prodromos
+resource "google_service_account" "prodromos_github" {
+  account_id   = "prodromos"
+  display_name = "Prodromos"
+  description  = "Terraform-managed service account for Prodromos terraform state bucket access"
+  project      = module.project_factory.project_id
+}
+
+resource "google_service_account_iam_binding" "prodromos_github" {
+  service_account_id = google_service_account.prodromos_github.name
+  role               = "roles/iam.workloadIdentityUser"
+  members            = ["principalSet://iam.googleapis.com/projects/${module.project_factory.project_id}/locations/global/workloadIdentityPools/${google_iam_workload_identity_pool.pool.id}/attribute.repository/${local.full_prodromos_repo}"]
+}
```

Comment:
My intention here is to limit impersonation of this service account to workloads running in GitHub actions on the `lsst-sqre/prodromos` repo.

Reply:
I was confused about how the principal in this binding maps to the provider, since it only refers to the pool. Turns out that the provider is specified in the GitHub workflow as an input to their `auth` action.